test/framework/functional/functional_test_cluster.py

```diff
@@ -300,6 +306,12 @@ def kill_master(self):
         :return: The killed master service with return code, stdout, and stderr set.
         :rtype: ClusterController
         """
+        # Delete testing database after we're done
+        if isfile(TEST_DB_NAME):
+            os.remove(TEST_DB_NAME)
+        else:
```

Comment:
If the tests are deterministic you should not need these conditionals. In case of a pre-mature test failure, you could just catch all errors in this method with at `try` block.